app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupMakingSettingsViewHolder.java

```diff
@@ -98,8 +100,16 @@ void afterGroupNumChanged() {
     // Given the number of names per group or number of groups, returns the corresponding number to use all the names
     private String getMatchingNumber(String inputtedNumber) {
         int newNumber = Integer.parseInt(inputtedNumber);
-        int offset = (int) Math.ceil((double) settings.getNameListSize() / newNumber);
-        return String.valueOf(offset);
+        double offset = (double) settings.getNameListSize() / newNumber;
+        checkGroupBalance(offset);
+        int matchingNumber = (int) Math.ceil(offset);
+        return String.valueOf(matchingNumber);
+    }
+
+    // check if the groups have the same names quantity
+    private void checkGroupBalance(double ratio) {
+        int warningMessageVisibility = ratio % 1 > 0 ? View.VISIBLE : View.GONE;
```

Comment:
From my understand, there's the list size and then the total number of names (# of groups * # of names per group). I think it's a legitimate use case to generate less names than the list size you have, something like generating 5 groups of 2 from a list of 60 names. I think we should only show the warning if the settings will use up all the names but not have enough.

The % 1 trick is quite nifty though, even if it's not the optimal way.

Reply:
if we do something like generating 5 groups of 2 from a list of 60 how should the autocomplete work, I think we should put a checkbox for autocomplete in the settings.